In [1]:
import pandas as pd
import numpy as np

# Customer Data

In [2]:
customer = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vR2YutHDy72Vd8M72466VQbcqbC3eBI7G_u186j1u3StHIXFoaxlhSuU3FBM9pyxkNOzUemefKGy78S/pub?output=csv')
customer.head()

CustomerID  Age  Gender Marital Status Income
0           1   55       1        Married   5,12
1           2   60       1        Married   6,23
2           3   32       1        Married   9,17
3           4   31       1        Married   4,87
4           5   58       1        Married   3,57

In [3]:
customer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 447 entries, 0 to 446
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   CustomerID      447 non-null    int64 
 1   Age             447 non-null    int64 
 2   Gender          447 non-null    int64 
 3   Marital Status  444 non-null    object
 4   Income          447 non-null    object
dtypes: int64(3), object(2)
memory usage: 17.6+ KB


In [4]:
customer.describe()

CustomerID         Age      Gender
count  447.000000  447.000000  447.000000
mean   224.000000   39.782998    0.458613
std    129.182042   12.848719    0.498842
min      1.000000    0.000000    0.000000
25%    112.500000   30.000000    0.000000
50%    224.000000   39.000000    0.000000
75%    335.500000   50.500000    1.000000
max    447.000000   72.000000    1.000000

In [5]:
customer.describe(include='object')

Marital Status Income
count             444    447
unique              2    369
top           Married      0
freq              340     16

Recommended Data Improvements:
1. Fix the income format (replace commas with dots if necessary to ensure numerical accuracy).
2. Handle missing marital status values, potentially using mode imputation.
3. Investigate age 0 entries to check for data entry errors.


# Store

In [6]:
store = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vQ4jHWECMy5IA4gzq27nziFaT7AFW5RxfU_7_XHPDGquI_i1ZbkzcKyNPK8xdXZR0Fx0dScjpGQH7xy/pub?output=csv')
store

StoreID         StoreName    GroupStore           Type  Latitude  \
0         1     Prima Tendean         Prima   Modern Trade  -6.20000   
1         2  Prima Kelapa Dua         Prima   Modern Trade  -6.91486   
2         3        Prima Kota         Prima   Modern Trade  -7.79707   
3         4       Gita Ginara          Gita  General Trade  -6.96667   
4         5           Bonafid          Gita  General Trade  -7.25045   
5         6            Lingga        Lingga   Modern Trade  -5.13540   
6         7       Buana Indah         Buana  General Trade   3.31669   
7         8     Sinar Harapan  Harapan Baru  General Trade   5.54829   
8         9            Lingga        Lingga   Modern Trade  -3.65470   
9        10      Harapan Baru  Harapan Baru  General Trade   3.59703   
10       11     Sinar Harapan      Prestasi  General Trade   5.33505   
11       12    Prestasi Utama      Prestasi  General Trade  -2.99093   
12       13             Buana         Buana  General Trade  -1.26916   
13       14          Priangan      Priangan   Modern Trade  -5.45000   

     Longitude  
0   106.816666  
1   107.608238  
2   110.370529  
3   110.416664  
4   112.768845  
5   119.423790  
6   114.590111  
7    95.323753  
8   128.190643  
9    98.678513  
10  101.447403  
11  104.756554  
12  116.825264  
13  105.266670

In [7]:
store.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14 entries, 0 to 13
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   StoreID     14 non-null     int64  
 1   StoreName   14 non-null     object 
 2   GroupStore  14 non-null     object 
 3   Type        14 non-null     object 
 4   Latitude    14 non-null     float64
 5   Longitude   14 non-null     float64
dtypes: float64(2), int64(1), object(3)
memory usage: 804.0+ bytes


In [8]:
store.describe()

StoreID   Latitude   Longitude
count  14.0000  14.000000   14.000000
mean    7.5000  -2.559441  109.463117
std     4.1833   4.958413    8.654790
min     1.0000  -7.797070   95.323753
25%     4.2500  -6.736145  104.884083
50%     7.5000  -4.395050  108.989384
75%    10.7500   2.170228  114.134794
max    14.0000   5.548290  128.190643

In [9]:
store.describe(include='object')

StoreName GroupStore           Type
count         14         14             14
unique        12          7              2
top       Lingga      Prima  General Trade
freq           2          3              8

# Product

In [10]:
product = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vQGu40ZuPzGEeO-MMLxdj6h3ecNRm1ScI-62y7VRTkUHYlMGJZ8_gA1LP1rSylxSrSeRk8GE7IHSwDi/pub?output=csv')
product

ProductID  Product Name  Price
0        P1     Choco Bar   8800
1        P2  Ginger Candy   3200
2        P3      Crackers   7500
3        P4   Potato Chip  12000
4        P5      Thai Tea   4200
5        P6        Cashew  18000
6        P7  Coffee Candy   9400
7        P8           Oat  16000
8        P9       Yoghurt  10000
9       P10  Cheese Stick  15000

In [11]:
product.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   ProductID     10 non-null     object
 1   Product Name  10 non-null     object
 2   Price         10 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 372.0+ bytes


In [12]:
product.describe()

Price
count     10.000000
mean   10410.000000
std     4890.455557
min     3200.000000
25%     7825.000000
50%     9700.000000
75%    14250.000000
max    18000.000000

In [13]:
product.describe(include='object')

ProductID Product Name
count         10           10
unique        10           10
top           P1    Choco Bar
freq           1            1

Recommended Data Improvements:
1. Further categorize products into types (e.g., snacks, beverages, dairy) for better segmentation.
2. Analyze price segmentation to identify pricing trends and potential product bundling opportunities.

# Transaction

In [14]:
transaction = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vRs4ibwaAdgnkNpJHQAu80O7nI55-AWxv6fZlB25TwrFDkUC1C3468DRlxvdscmD45UpMT0yV4AvgEu/pub?output=csv')
transaction.head()

TransactionID  CustomerID        Date ProductID  Price  Qty  TotalAmount  \
0       TR11369         328  01/01/2022        P3   7500    4        30000   
1       TR16356         165  01/01/2022        P9  10000    7        70000   
2        TR1984         183  01/01/2022        P1   8800    4        35200   
3       TR35256         160  01/01/2022        P1   8800    7        61600   
4       TR41231         386  01/01/2022        P9  10000    1        10000   

   StoreID  
0       12  
1        1  
2        4  
3        4  
4        4

In [15]:
transaction.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5020 entries, 0 to 5019
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   TransactionID  5020 non-null   object
 1   CustomerID     5020 non-null   int64 
 2   Date           5020 non-null   object
 3   ProductID      5020 non-null   object
 4   Price          5020 non-null   int64 
 5   Qty            5020 non-null   int64 
 6   TotalAmount    5020 non-null   int64 
 7   StoreID        5020 non-null   int64 
dtypes: int64(5), object(3)
memory usage: 313.9+ KB


In [16]:
transaction.describe()

CustomerID         Price          Qty   TotalAmount      StoreID
count  5020.000000   5020.000000  5020.000000   5020.000000  5020.000000
mean    221.263745   9684.800797     3.644622  32279.482072     7.489841
std     129.672955   4600.708780     1.855295  19675.462455     4.028502
min       1.000000   3200.000000     1.000000   7500.000000     1.000000
25%     108.000000   4200.000000     2.000000  16000.000000     4.000000
50%     221.000000   9400.000000     3.000000  28200.000000     7.000000
75%     332.000000  15000.000000     5.000000  47000.000000    11.000000
max     447.000000  18000.000000    10.000000  88000.000000    14.000000

In [17]:
transaction.describe(include='object')

TransactionID        Date ProductID
count           5020        5020      5020
unique          4908         365        10
top          TR71313  02/03/2022        P5
freq               3          31       814

Recommended Data Improvements:
1. Check for duplicate transactions (since there are 4,908 unique TransactionIDs but 5,020 rows).
2. Analyze product demand trends to see which items drive the most sales.
3. Investigate high-value transactions to understand customer purchasing behavior.
Segment transactions by store to identify top-performing locations.


# Join Data

In [18]:
# Menyesuaikan nama kolom jika perlu
customer.columns = ["customer_id", "age", "gender", "marital_status", "income"]
store.columns = ["store_id", "store_name", "group_store", "type", "latitude", "longitude"]
product.columns = ["product_id", "product_name", "price"]
transaction.columns = ["transaction_id", "customer_id", "transaction_date", "product_id", "price", "qty", "total_amount", "store_id"]

# Convert income ke format float jika ada koma
customer["income"] = customer["income"].astype(str).str.replace(",", ".").astype(float)

# Convert transaction_date ke format datetime
transaction["transaction_date"] = pd.to_datetime(transaction["transaction_date"], format="%d/%m/%Y")

# Melakukan join data
merged_df = transaction \
    .merge(customer, on="customer_id", how="left") \
    .merge(product, on="product_id", how="left") \
    .merge(store, on="store_id", how="left")

# Simpan hasil ke CSV
merged_df.to_csv("kalbe_data.csv", index=False, encoding="utf-8")

print("Data berhasil digabung dan disimpan dalam 'kalbe_data.csv'")


Data berhasil digabung dan disimpan dalam 'kalbe_data.csv'
